### ISSUE TITLE
Vale does not detect errors

### ISSUE THREAD

**Issue by ariebonn** [jdkato/txtlint#999]
### Check for existing issues

- [x] Completed

### Environment

OS: Ubuntu 22.04.5 LTS
Install method: sudo snap install vale
Vale version:  v3.11.2

### Describe the bug / provide steps to reproduce it

I installed Vale and am having trouble getting it to work. First issue is that it only works with a .vale.ini file from `/home/arie/.config/vale`. If I place the Vale file in my repo root it doesn't find it, even if I specify the `--config` option. It's weird because the file is there, but it says that it doesn't exist.

Anyways, I figured that it works when the Vale file is in the home folder so I tried to test it with an MD file. This is my config:

```
StylesPath = styles
MinAlertLevel = suggestion
Packages = RedHat, proselint
[*]
BasedOnStyles = Vale, RedHat, proselint, Test
```
I executed `vale sync` and verified it downloaded the styles.

I created a custom style in the Test folder:

```
extends: substitution
message: Consider using '%s' instead of '%s'
link: 'www.internal-tools.io'
level: warning
ignorecase: false
action:
 name: replace
# swap maps tokens in the form of bad: good
swap:
   applications: apps
   config: configuration
```

And created the following MD file:

```
# Hello World!?!

Hello world world. You can use your internal tools too build your own production grade applications config.
```

When I execute Vale on the file it outputs:

```
✔ 0 errors, 0 warnings and 0 suggestions in stdin.
```
Would appreciate any help in determining what I'm doing wrong.

**Comment by jdkato** [comment]
Thanks for the detailed report — both symptoms come from snap's **strict confinement**, which limits where Vale can read files.

What the snap can see:
- Files under your home directory (`$HOME`) — via the auto-connected `home` interface.
- `~/.vale.ini`, `~/.config/vale`, and `~/.local/share/vale` — via a bundled personal-files plug (that's why the config in `~/.config/vale` worked).
- USB/other mounts — only after `sudo snap connect vale:removable-media`.

So the two things you saw:

1. **Repo-root config "is there but doesn't exist."** If your repo lives *outside* `$HOME` (e.g. `/srv`, `/opt`, `/data`, a mount, etc.), the snap can't read it — even with `--config` — so it reports the file missing.
2. **"0 errors … in stdin."** When `vale file.md` can't see `file.md`, it falls back to treating the argument as a literal string to lint, so it lints the text `"file.md"`, finds nothing, and reports `in stdin`.

A couple of questions / fixes:
- **Where does your repo live?** If it's under your home directory, things should work out of the box — please share the path. If it's elsewhere, that's the cause.
- For a repo on a mount/removable drive: `sudo snap connect vale:removable-media`.
- For repos in arbitrary locations (`/srv`, `/opt`, …) that no snap interface covers, use the non-sandboxed binary instead — see https://vale.sh/docs/install.

Your config and `Test` rule look correct, so once Vale can actually read the directory it'll pick everything up.
